### PR TITLE
Term startup fixes

### PR DESCRIPTION
**Pull request by j-jasik**
This PR is all about smoothing out the term startup process. The following things were accomplished:

1. On term startup I confirmed that the new semester is immediately visible and all students are soft deleted. We used to try to demote tutors too, but that function has been abandoned.

2. Courses weren't being pulled in properly but now they are and it's nice and smooth.

3. The checkboxes to view courses were bugged, but now they are only checked if the course is offered for a schedule or session.

4. There is also a bug fix in here to double check for an existing user in the system on user creation.

This has been tested extensively locally and on xp to ensure as few issues as possible for the admins.

### CODE REVIEW

File: sciencelabs/__init__.py [hunk 1]
```diff
@@ -15,9 +15,9 @@
 from sciencelabs.db_repository.user_functions import User
 from sciencelabs.db_repository.schedule_functions import Schedule
 
-sentry = Sentry(app, dsn=app.config['SENTRY_URL'], logging=True, level=logging.INFO)
-if app.config['ENVIRON'] == 'prod':
-    from sciencelabs import error
+# sentry = Sentry(app, dsn=app.config['SENTRY_URL'], logging=True, level=logging.INFO)
+# if app.config['ENVIRON'] == 'prod':
+#     from sciencelabs import error
```
Comment (c-schwarze) [inline sciencelabs/__init__.py]: probably want this to stay in

File: sciencelabs/db_repository/course_functions.py [hunk 2]
```diff
@@ -163,8 +164,15 @@ def create_coursecode(self, cc_info):
         db_session.commit()
 
     def check_for_existing_course(self, c_info):
+        term, year, *rest = (c_info['term'].split('-')[0].split(' '))
+        semester_id = None
+        semester_list = flask_session['SEMESTER-LIST']
+        for semesters in semester_list:
+            if semesters['year'] == year and semesters['term'] == term:
+                semester_id = semesters['id']
```
Comment (c-schwarze) [inline sciencelabs/db_repository/course_functions.py]: 172 - can probably break once you found it?

File: sciencelabs/db_repository/course_functions.py [hunk 3]
```diff
@@ -163,8 +164,15 @@ def create_coursecode(self, cc_info):
         db_session.commit()
 
     def check_for_existing_course(self, c_info):
+        term, year, *rest = (c_info['term'].split('-')[0].split(' '))
```
Comment (c-schwarze) [inline sciencelabs/db_repository/course_functions.py]: 167 - seems like a weird way to check if a course exists by doing a split, 0 index, and split?

File: sciencelabs/__init__.py (hunk 1 above)
Comment (j-jasik) [reply sciencelabs/__init__.py]: Yes! fixed.

File: sciencelabs/db_repository/course_functions.py (hunk 2 above)
Comment (j-jasik) [reply sciencelabs/db_repository/course_functions.py]: good call - fixed.

**Pull request merged**